### ISSUE TITLE
Consider non-LGPL character encoding library

### ISSUE THREAD

**Issue by johnthagen** [encode/httpx#1013]
Python binary applications cannot legally depend on `httpx` because of it's dependency on `chardet`.

https://github.com/encode/httpx/blob/8c84210555294ce10c681edae2df71627095a4e9/setup.py#L61

## Describe the solution you would like.

Here is a current license summary of `httpx` and its dependencies:

```
 % pip-licenses --from=mixed
 Name         Version     License                                             
 certifi      2020.4.5.1  Mozilla Public License 2.0 (MPL 2.0)                
 chardet      3.0.4       GNU Library or Lesser General Public License (LGPL) 
 h11          0.9.0       MIT License                                         
 h2           3.2.0       MIT License                                         
 hpack        3.0.0       MIT License                                         
 hstspreload  2020.5.30   BSD License                                         
 httpcore     0.9.1       BSD License                                         
 httpx        0.13.3      BSD License                                         
 hyperframe   5.2.0       MIT License                                         
 idna         2.9         BSD License                                         
 rfc3986      1.4.0       Apache Software License                             
 sniffio      1.1.0       MIT License, Apache Software License
```

`chardet` is [LGPL licensed](https://opensource.org/licenses/LGPL-3.0). I'm not a lawyer, but my understanding is that this makes it unsuitable for including inside of a binary due to the restriction that users need to be able to have the ability to swap out the LGPL library for another of their choosing, something infeasible for a binary distribution.

Users who use `httpx` and want to deploy via [Pyinstaller](https://www.pyinstaller.org/), [Nuitka](https://nuitka.net/), [PyOxdizer](https://pyoxidizer.readthedocs.io/en/stable/), etc. therefore are unable to use `httpx`, as I understand it.

[`charset_normalizer`](https://github.com/ousret/charset_normalizer) is one such alternative that is MIT licensed, thus making it appropriate for binary deployments.

**Comment by tomchristie** [comment]
I'm not seeing anything in https://github.com/chardet/chardet/blob/master/LICENSE that's obviously problematic with how *we* distribute httpx. Also similar comments in issues around https://github.com/psf/requests/search?q=lgpl&type=Issues

I'd be okay with reviewing any PR that made `chardet` an optional import, which gives alternate packagers os other options, but we'd still want *our* PyPI packaging to include it.

**Comment by johnthagen** [comment]
This was fixed in #1269
